diff -rNu lightning-1.2pre/lightning/i386/fp.h lightning-1.2pre-seam-patches/lightning/i386/fp.h
--- lightning-1.2pre/lightning/i386/fp.h	2004-03-15 11:55:48.000000000 +0100
+++ lightning-1.2pre-seam-patches/lightning/i386/fp.h	2004-04-16 10:00:30.000000000 +0200
@@ -242,21 +242,21 @@
         jit_fxch ((rs), FISTPLm(0, _ESP, 0, 0)),       \
 	POPLr((rd)))
 
-#define jit_fp_test(d, s1, s2, n, and, res)            \
+#define jit_fp_test(d, s1, s2, n, _and, res)            \
        (((s1) == 0 ? FUCOMr((s2)) : (FLDr((s1)), FUCOMPr((s2) + 1))),                          \
         ((d) != _EAX ? MOVLrr(_EAX, (d)) : 0),                 \
         FNSTSWr(_EAX),                                         \
         SHRLir(n, _EAX),                                       \
-        ((and) ? ANDLir((and), _EAX) : MOVLir(0, _EAX)),       \
+        ((_and) ? ANDLir((_and), _EAX) : MOVLir(0, _EAX)),       \
         res,                                                   \
         ((d) != _EAX ? _O (0x90 + ((d) & 7)) : 0))     /* xchg */
 
-#define jit_fp_btest(d, s1, s2, n, and, cmp, res)                      \
+#define jit_fp_btest(d, s1, s2, n, _and, cmp, res)                      \
        (((s1) == 0 ? FUCOMr((s2)) : (FLDr((s1)), FUCOMPr((s2) + 1))),                          \
         PUSHr(_EAX),                                           \
         FNSTSWr(_EAX),                                         \
         SHRLir(n, _EAX),                                       \
-        ((and) ? ANDLir ((and), _EAX) : 0),                    \
+        ((_and) ? ANDLir ((_and), _EAX) : 0),                    \
         ((cmp) ? CMPLir ((cmp), _AL) : 0),                     \
         POPr(_EAX),                                            \
         res ((d)))
