--- stockhausen-devel/bin/alicerun	2003-02-12 16:51:38.000000000 +0100
+++ alice-0.9.1/bin/alicerun	2003-02-13 09:55:57.000000000 +0100
@@ -52,7 +52,7 @@
     ALICE_LOAD=$ALICE_LOAD${sep}$ALICE_LOAD_SUFFIX
 fi
 
-: ${OZ_LOAD:=cache=~/.oz/cache${sep}cache=$OZHOME/cache}
+: ${OZ_LOAD:=cache=~/.oz/cache${sep}cache=/usr/lib/mozart/cache${sep}cache=/usr/share/mozart/cache}
 
 OZ_LOAD=$ALICE_LOAD${sep}pattern=x-alice:/'?{x}'=$ALICE_HOME/'?{x}'.ozf${sep}$OZ_LOAD
 
